problemtools/verifyproblem.py

```diff
@@ -310,6 +310,23 @@ def check(self, args):
             if field not in TestCaseGroup._DEFAULT_CONFIG.keys():
                 self.warning("Unknown key '%s' in '%s'" % (field, os.path.join(self._datadir, 'testdata.yaml')))
 
+        if self._problem.config.get('type') == 'pass-fail':
+            for key in TestCaseGroup._SCORING_ONLY_KEYS:
+                if self.config.get(key) is not None:
+                    self.error("Key '%s' is only applicable for scoring problems, this is a pass-fail problem" % key)
+
+        if self._problem.config.get('type') == 'scoring':
+            # Check grading
+            try:
+                score_range = self.config['range']
+                (min_score, max_score) = map(float, score_range.split())
+                if min_score >= max_score:
+                    self.error("Invalid score range '%s': minimum score must be smaller than maximum score" % score_range)
+            except VerifyError:
+                raise
+            except:
+                self.error("Invalid format '%s' for grading.range: must be exactly two floats" % score_range)
```

Comment:
I think there are some checks missing here.  At the very least, should check that on_reject is break or continue.  Also, what happens if I don't specify all the values in `_SCORING_ONLY_KEYS` (for a scoring problem)?